File: JAVA/2807. Insert Greatest Common Divisors in Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    private int gcd(int a,int b)
    {
        return (b==0?a:gcd(b,a%b));
    }
    public ListNode insertGreatestCommonDivisors(ListNode head) {
        ListNode temp=head;
        while(temp!=null && temp.next!=null){
            ListNode t=new ListNode(gcd(temp.val,temp.next.val));
            t.next=temp.next;
            temp.next=t;
            temp=temp.next.next;
        }
        return head;
    }
}

File: C++/2807. Insert Greatest Common Divisors in Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int gcd(int a,int b){
        return(b==0?a:gcd(b,a%b));
    }
    ListNode* insertGreatestCommonDivisors(ListNode* head) {
        ListNode* temp=head;
        while(temp&&temp->next){
            ListNode* t = new ListNode(gcd(temp->val, temp->next->val));
            t->next=temp->next;
            temp->next=t;
            temp=temp->next->next;
        }
        return head;
    }
};